### DISCUSSION TITLE
Fetch failed: AggregateError [ETIMEDOUT] on Node.js 20

### DISCUSSION THREAD

**Discussion by raky291** [vercel/next.js#70423] [Help]
### Link to the code that reproduces this issue

https://github.com/raky291/pokedex

### To Reproduce

1. Create a Next.js project with Docker (feel free to use my repo)
2. Change Node version to 20 on Dockerfile
3. Run the application and make fetch requests to an API

### Current vs. Expected behavior

I'm expecting it to perform fetch requests correctly like it does in the development (next dev) or production (next start) environment but it throws errors.

![image](https://github.com/user-attachments/assets/d42ae24c-92d0-4056-8dba-b44b5634acee)

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
  Available memory (MB): 32525
  Available CPU cores: 20
Binaries:
  Node: 20.16.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.11 // Latest available version is detected (14.2.11).
  eslint-config-next: 14.2.11
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.5.4
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Output (export/standalone)

### Which stage(s) are affected? (Select all that apply)

Other (Deployed)

### Additional context

I have Node version 20 installed on my machine. In development (next dev) and production (next start) it works fine. But with Docker (standalone) it throws errors. It's weird because with version 20 in Docker if you make a single fetch request it works.

**I had to downgrade the Node version to 18 to fix the issue.**

**Comment by samcx** [comment]
@raky291 Seems related to Node → https://github.com/nodejs/undici/issues/2990

**Comment by gregonarash** [comment]
@raky291  @samcx  I believe this is linked to underlying Node default network-family-autoselection-attempt-timeout

https://github.com/nodejs/undici/issues/2990

On some networks (like e.g. mine today on a LTE tethered connection in a country far away from my database provider) the default network-family-autoselection-attempt-timeout 250 ms to resolve domain IP might not be enough and results in ETIMEDOUT.

Doubling that time solves all issues for me:
export NODE_OPTIONS="--network-family-autoselection-attempt-timeout=500"

**Reply by vemman** [reply]
Thanks this worked for me I am also on a similar setup ie behind a 5G router!
